lib/vrl/tests/src/main.rs

```diff
@@ -0,0 +1,292 @@
+use ansi_term::Colour;
+use chrono::{DateTime, Utc};
+use glob::glob;
+use vrl::{diagnostic::Formatter, state, Runtime, Value};
+use std::fs;
+use std::path::Path;
+use std::str::FromStr;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "VRL Tests", about = "Vector Remap Language Tests")]
+pub struct Cmd {
+    #[structopt(short, long)]
+    pattern: Option<String>,
+
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+fn main() {
+    let cmd = Cmd::from_args();
+
+    let mut failed_count = 0;
+    let mut category = "".to_owned();
+    for entry in glob("tests/**/*.vrl").expect("valid pattern") {
+        let path = match entry {
+            Ok(path) => path,
+            Err(_) => continue,
+        };
+
+        if &path.to_string_lossy() == "tests/example.vrl" {
+            continue;
+        }
+
+        if let Some(pat) = &cmd.pattern {
+            if !path.to_string_lossy().contains(pat) {
+                continue;
+            }
+        }
+
+        let test_category = test_category(&path);
+        if category != test_category {
+            category = test_category;
+            println!("{}", Colour::Fixed(3).bold().paint(format!("{}", category)));
+        }
+
+        let name = test_name(&path);
+        let dots = 60 - name.len();
+
+        print!("  {}{}", name, Colour::Fixed(240).paint(".".repeat(dots)));
+
+        let mut test = match Test::new(&path) {
+            Ok(test) => test,
+            Err(err) => {
+                println!("{}", Colour::Purple.bold().paint("INVALID"));
+                println!("{}", Colour::Red.paint(err));
+                failed_count += 1;
+
+                continue;
+            }
+        };
+
+        if test.skip {
+            println!("{}", Colour::Yellow.bold().paint("SKIPPED"));
+        }
+
+        let state = state::Runtime::default();
+        let mut runtime = Runtime::new(state);
+        let program = vrl::compile(&test.source, &stdlib::all());
+
+        let want = test.result;
+
+        match program {
+            Ok(program) => {
+                let result = runtime.resolve(&mut test.object, &program);
+
+                match result {
+                    Ok(got) => {
+                        if !test.skip {
+                            let want = if want.starts_with("r'") {
```

Comment:
Would check `want.ends_with("'")` here as well.

Reply:
Thanks for the suggestion. As mentioned [here](https://github.com/timberio/vector/pull/6353#discussion_r570541767) I didn't spend too much time making this entirely fool-proof, but I'll make the suggested change 👍